# backtest-case: PT-032
# kind: case
# base: 42b840051647eef89761a16dfdff87e82bb253ab
# derived: diff(BASE:transformer_engine/pytorch/__init__.py, PRE:transformer_engine/pytorch/__init__.py)
--- a/transformer_engine/pytorch/__init__.py
+++ b/transformer_engine/pytorch/__init__.py
@@ -1,3 +1,5 @@
+# This file was modified for portability to AMDGPU
+# Copyright (c) 2026, Advanced Micro Devices, Inc. All rights reserved.
 # Copyright (c) 2022-2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 #
 # See LICENSE for license information.
@@ -47,6 +49,7 @@
 from transformer_engine.pytorch.quantization import is_mxfp8_available
 from transformer_engine.pytorch.quantization import is_fp8_block_scaling_available
 from transformer_engine.pytorch.quantization import is_nvfp4_available
+from transformer_engine.pytorch.quantization import is_mxfp4_available
 from transformer_engine.pytorch.quantization import get_default_recipe
 from transformer_engine.pytorch.utils import get_cudnn_version
 from transformer_engine.pytorch.utils import get_device_compute_capability
@@ -63,10 +66,12 @@
 from transformer_engine.pytorch import optimizers
 from transformer_engine.pytorch.export import onnx_export
 from transformer_engine.pytorch.cross_entropy import parallel_cross_entropy
-from transformer_engine.pytorch.newton_schulz import (
-    CusolverMpCtx,
-    newton_schulz,
-)
+from torch.utils.cpp_extension import IS_HIP_EXTENSION as _IS_HIP_EXTENSION
+if not _IS_HIP_EXTENSION:
+    from transformer_engine.pytorch.newton_schulz import (
+        CusolverMpCtx,
+        newton_schulz,
+    )
 from transformer_engine.pytorch.quantized_tensor import QuantizedTensorStorage
 from transformer_engine.pytorch.quantized_tensor import QuantizedTensor
 from transformer_engine.pytorch.quantized_tensor import Quantizer
